packages/nimble-components/src/rich-text-editor/index.ts

```diff
@@ -347,8 +495,100 @@ export class RichTextEditor extends FoundationElement {
                 return false;
         }
     }
+
+    private unbindEditorUpdateEvent(): void {
+        this.tiptapEditor.off('update');
+    }
+
+    /**
+     * input event is fired when there is a change in the content of the editor.
+     *
+     * https://tiptap.dev/api/events#update
+     */
+    private bindEditorUpdateEvent(): void {
+        this.tiptapEditor.on('update', () => {
+            this.$emit('input');
+            this.queueUpdateScrollbarWidth();
+            this.queueUpdateEditorHeight();
+        });
+    }
+
+    private queueUpdateScrollbarWidth(): void {
+        if (!this.$fastController.isConnected) {
+            return;
+        }
+        if (!this.updateScrollbarWidthQueued) {
+            this.updateScrollbarWidthQueued = true;
+            DOM.queueUpdate(() => this.updateScrollbarWidth());
+        }
+    }
+
+    private updateScrollbarWidth(): void {
+        this.updateScrollbarWidthQueued = false;
+        this.scrollbarWidth = this.editor.offsetWidth - this.editor.clientWidth;
+    }
+
+    private queueUpdateEditorHeight(): void {
+        if (!this.$fastController.isConnected) {
+            return;
+        }
+        if (!this.updateEditorHeightQueued) {
+            this.updateEditorHeightQueued = true;
+            DOM.queueUpdate(() => this.updateEditorHeight());
+        }
+    }
+
+    private updateEditorHeight(): void {
+        this.updateEditorHeightQueued = false;
+        this.updateEditorMinMaxHeight();
+    }
+
+    private onResize(): void {
+        this.scrollbarWidth = this.editor.offsetWidth - this.editor.clientWidth;
+        this.updateEditorMinMaxHeight();
+    }
+
+    private updateEditorMinMaxHeight(): void {
```

Comment:
We really try hard to avoid encoding style values and doing style calculations in TS code. The CSS layout engine is optimized to do this for us so we'd rather rely on it than on JS/TS code, which is slower and less easy to read. I would like us to try to encode this logic in CSS rules rather than calling `getComputedStyle` and doing pixel calculations.

Another concern is about who controls the min and max height of the editor. I was expecting that to be something provided by clients by setting CSS rules on the text editor but I see some hardcoded values here that indicate it's intrinsic to the editor.

I don't know enough about the constraints that TipTap applies to offer more concrete suggestions or to know if these suggestions are possible. If you'd like to brainstorm more, we could set up a call with me and @m-akinc (I'm free on Tuesday after the UI WG). Or if there's time pressure for this PR we could consider removing the sizing code for now and splitting it into a follow up PR.

Reply:
I have removed the entire logic from the TS code and modified the CSS styles to achieve what we wanted. Please review the new implementation and let us know your feedback.

Reply:
Other than the feedback Milan mentioned about `inherit` and the resize jitter I mentioned in the other thread, I think the new approach is looking nice.

Reply:
Thanks for the feedback. As I mentioned in the other comment, we will be moving this to a follow-up PR. Therefore, I am closing this comment.